File: dev-docs/snippets/events/sync.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Fires when a document's sync state with the server flips.
export function sync(client: JsBaoClient) {
  // #region example
  client.on("sync", (payload) => {
    console.log(payload.documentId, "synced:", payload.synced);
  });
  // #endregion example
}

File: dev-docs/snippets/events/sync.swift
import JsBaoClient

// Fires when a document's sync state with the server flips.
func sync(client: JsBaoClient) {
  // #region example
  let sub = client.events.on(.sync) { (event: SyncEvent) in
    print(event.documentId, "synced:", event.synced)
  }
  // #endregion example
  _ = sub
}
